pyproject.toml

```diff
@@ -1,6 +1,6 @@
 [build-system]
 requires = [
-    "Cython",
+    "Cython>=3.1.0a0",
```

Comment:
If we are OK with putting this in the build system area to a high pin, we can automatically get pip to install the required Cython for us without needing to update any of the cibuildwheel BEFORE_BUILD steps.

Note that we can still be built with `Cython<3.1`, for example if you install with `pip install --no-build-isolation .` and have a lower Cython installed in your environment.